### ISSUE TITLE
[HOLD for payment 2022-05-24][$1000] Native Apps - There is an extra space above codeblocks in android and IOS - reported by @sobitneupane

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8089]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Send codeblock with message above and below: 
    Here is come code
    ```
    const myLineOfCode = 'some code';
   ```

## Expected Result:
Same output in all platforms.

## Actual Result:
Extra space above codeblocks in android and IOS.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platform:

Where is this issue occurring?

- iOS
- Android

**Version Number:** 1.1.41-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation  
Related to https://github.com/Expensify/App/issues/7496#issuecomment-1063237609 

![Screenshot_1644495649](https://user-images.githubusercontent.com/43995119/157807679-ad55c965-109f-4206-81dd-63af382b7c14.png)
<img width="1440" alt="Screen Shot 2022-02-10 at 18 04 02" src="https://user-images.githubusercontent.com/43995119/157807680-8874382e-c808-48a1-b67b-749c13ed8c73.png">

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @sobitneupane
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1644507497800049

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MonilBhavsar (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MonilBhavsar** [comment]
Looks like `<Text>` component is occupying more than half of height of `<ReportActionItemGrouped>` component

<img width="360" alt="Screenshot 2022-03-14 at 9 44 26 PM" src="https://user-images.githubusercontent.com/32012005/158215134-f30a9d9b-5bab-46c8-b13b-3d5ea30fc5f5.png">

Not blocking anything and occurs on iOS/Android, so low priority

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcochavezf (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by JmillsExpensify** [comment]
Still waiting for volunteers. Increased the Upwork job to $500 per the process.

**Comment by JmillsExpensify** [comment]
Still waiting on volunteers. Increasing the price to $1000 and adding to the issue title.

**Comment by aneequeahmad** [comment]
### **PROPOSAL**


**CAUSE OF ISSUE**

Converting the comment from MD into HTML in addAction function(Report.js) is causing the issue. ExpensiMark.js replace function is called with message text that applies rules to text to ensure that any Html the user puts into the comment field shows as raw Html. In the rules below rule
 ` {
                // when <pre> and <br /> occur together extra line is added so removing <br />
                name: 'replacepre',
                regex: /<\/pre>\s*<br\s*[/]?>/gi,
                replacement: '</pre>',
     },`
    is not replacing <br /><pre> with <pre> to remove extra line added but only removes <pre>.
    
    **BEST POSSIBLE SOLUTION**
    
    This rule when added
     `{
                //  when <br /> and <pre> occur together extra line is added remove <br />
                name: 'replacebr',
                regex: /<br\s*[\/]?><pre>\s*/gi,  
                replacement: '<pre>'
            }`
replaces <br /><pre> with <pre> which fixes the issue. 

**TO BE NOTED AND NEED SUGGESTION**
Also this regex  /<\/pre>\s*<br\s*[/]?>/gi doesn't seem to be valid so replacing it with `/<\/pre>\s*<br\s*[\/]?>/gi` to make this regex valid.

ScreenShot is attached after the fix is applied.

<img width="377" alt="Screen Shot 2022-04-07 at 5 00 45 AM" src="https://user-images.githubusercontent.com/36359331/162093809-c7a672d8-6098-4994-bd75-f22a0bf4fe9d.png">

**Comment by JmillsExpensify** [comment]
@rushatgabhane Circling back on this one because we owe feedback for this proposal. Thanks!

**Comment by rushatgabhane** [comment]
@aneequeahmad Sorry for the extreme delay. Your proposal is pretty much spot on!
```
{
      //  when <br /> and <pre> occur together extra line is added remove <br />
      name: 'replacebr',
      regex: /<br\s*[\/]?><pre>\s*/gi,  
      replacement: '<pre>'
}
```
> Also this regex /<\/pre>\s*<br\s*[/]?>/gi doesn't seem

And yes, you're right, this regex is wrong. Let's fix that too.

**Comment by rushatgabhane** [comment]
🎀 👀 🎀 C+ reviewed 
@marcochavezf I like @aneequeahmad's [proposal](https://github.com/Expensify/App/issues/8089#issuecomment-1090940503).

Let's get the party started 👯‍♀️

**Comment by JmillsExpensify** [comment]
Checking in on next steps here? Are we waiting on @marcochavezf for agreement, or are we ready to start working on the PR?

**Comment by rushatgabhane** [comment]
> Are we waiting on @marcochavezf for agreement

Yep, waiting for @marcochavezf's review

**Comment by marcochavezf** [comment]
Thank you guys! Assigning @aneequeahmad

**Comment by aneequeahmad** [comment]
Applied on Upwork, PR is raised for Expensify-common. 

cc: @marcochavezf, @rushatgabhane

**Comment by aneequeahmad** [comment]
Also, I think the price should be doubled ($2000) hehe, as I have pointed out another bug in the already written code and fixed that too. Let me know if it makes sense or not(I read somewhere if I feel like the price should be increased I should mention it in the comment).

cc: @marcochavezf

**Comment by marcochavezf** [comment]
Hi @aneequeahmad, what other issue did you fix? Could you point me where you read that the price should be increased?

**Comment by aneequeahmad** [comment]
Hi @marcochavezf, the other issue was there was the other regex rule doesn't seem valid which i pointed out above in my proposal [here](https://github.com/Expensify/App/issues/8089#:~:text=Also%20this%20regex%20/%3C%5C/pre%3E%5Cs*%3Cbr%5Cs*%5B/%5D%3F%3E/gi%20doesn%27t%20seem%20to%20be%20valid%20so%20replacing%20it%20with%20/%3C%5C/pre%3E%5Cs*%3Cbr%5Cs*%5B%5C/%5D%3F%3E/gi%20to%20make%20this%20regex%20valid.) . I read about the reevaluation of price [here](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#:~:text=If%20you%20think%20your%20compensation%20should%20be%20increased%20for%20a%20specific%20job%2C%20you%20can%20request%20a%20reevaluation%20by%20commenting%20in%20the%20Github%20issue%20where%20the%20Upwork%20job%20was%20posted.)

**Comment by marcochavezf** [comment]
Oh I see, thanks @aneequeahmad, well we don't know which problem in the app is solving that suggestion, also the current regex `/<\/pre>\s*<br\s*[/]?>/gi` is already valid for ` </pre><br/>` and ` </pre><br>`, no?

**Comment by aneequeahmad** [comment]
@marcochavezf this [regex](https://github.com/Expensify/App/issues/8089#:~:text=And%20yes%2C%20you%27re%20right%2C%20this%20regex%20is%20wrong.%20Let%27s%20fix%20that%20too.) is not valid as seen in the conversation. as the problem is eslint throws error for this regex `/<\/pre>\s*<br\s*[\/]?>/gi` have to add disable rule for these regex.

**Comment by marcochavezf** [comment]
Commented in the [PR](https://github.com/Expensify/expensify-common/pull/442), the eslint error is thrown by the escape `/` character, which is not needed inside of `[ ]` in JS

**Comment by marcochavezf** [comment]
Requested a few changes in the PR, it's still in progress

**Comment by aneequeahmad** [comment]
@JmillsExpensify, The Upwork [job](https://www.upwork.com/jobs/~0127315d24f0e23219) is no longer available for some reason. PR is merged and PR for App is raised as well. Thanks

**Comment by JmillsExpensify** [comment]
No worries. I've got you covered here: https://www.upwork.com/jobs/~012ed1a030f60ce0f5. Can you apply? Then we can issue payment once the standard regression period ends.

**Comment by JmillsExpensify** [comment]
cc @sobitneupane and @rushatgabhane as well for the new job posting.

**Comment by aneequeahmad** [comment]
@JmillsExpensify Applied, yeah sounds good let's wait for the regression period. Thanks for your message.

**Comment by JmillsExpensify** [comment]
Still working through considerations in the last linked PR.

**Comment by JmillsExpensify** [comment]
Working through issues in the linked PR to get the project running. No progress since last week.

**Comment by rushatgabhane** [comment]
Interesting..

Looks like @aneequeahmad's [changes to `expensify-common`](https://github.com/Expensify/expensify-common/pull/442) hit production 3 days ago because of an another PR https://github.com/Expensify/App/pull/8948, which updated `expensify-common` to a more up-to-date version.

If @marcochavezf agrees with me as well, we can ask the QA team to test this issue.

**Comment by marcochavezf** [comment]
Yeah, it's interesting that `expensify-common` was eventually updated in E/App because of another PR. Thanks for testing again on Android @rushatgabhane!

>If @marcochavezf agrees with me as well, we can ask the QA team to test this issue and close it out in 7 days.

Sure, sounds like a good plan to me! I already [sent a message](https://expensify.slack.com/archives/C9YU7BX5M/p1653067928335069) to the QA team to test again.

**Comment by mvtglobally** [comment]
@marcochavezf 
Can you confirm the build to test this on?

**Comment by rushatgabhane** [comment]
@mvtglobally production build on mobile platforms.

**Comment by JmillsExpensify** [comment]
Thanks everyone! So based on when https://github.com/Expensify/App/pull/8948 hit production, today is the end of the regression period. Accordingly, I'm holding to hold for the next QA test. Assuming that is happening sometime today, so @mvtglobally it'd be great if you can confirm when this passes. I'll go ahead and put a hold for payment tomorrow.

**Comment by mvtglobally** [comment]
@JmillsExpensify @rushatgabhane
It seems like there is still some gap and issue is repro. See below 
![ios_8089-2105](https://user-images.githubusercontent.com/43995119/169950203-eb2edbee-9e3a-48c6-9b5b-41458469a234.jpg)

**Comment by rushatgabhane** [comment]
@mvtglobally all those look like they're different messages.

Can we try that again with a single message and see if it works?
![Screenshot_20220524-081339](https://user-images.githubusercontent.com/29673073/169954068-e1451957-eb7d-47da-b851-2adf166186af.png)

**Comment by aneequeahmad** [comment]
@JmillsExpensify The Upwork job is no longer available, also i haven't been hired for this job. 
Could i ask when the payment will be issued ? I'm unaware of the process as this is my first job. Thanks

**Comment by rushatgabhane** [comment]
@aneequeahmad the job might have expired and a new one will be created. 

> Could i ask when the payment will be issued

Once the issue is not reproducible by the QA team, the payments will be settled.
Thanks for understanding!

**Comment by aneequeahmad** [comment]
@rushatgabhane Ahhh alright! thank you for your comment 🥇

**Comment by JmillsExpensify** [comment]
> @JmillsExpensify The Upwork job is no longer available, also i haven't been hired for this job. Could i ask when the payment will be issued ? I'm unaware of the process as this is my first job. Thanks

You're good! You already applied to the existing job post here: https://www.upwork.com/jobs/~012ed1a030f60ce0f5. I've hired you, same for @rushatgabhane as C+ and @sobitneupane for reporting. Thanks all!

**Comment by JmillsExpensify** [comment]
@mvtglobally I just tested this on the latest build and I think we're good. Would you mind confirming so we can close this out in the next QA round? Thanks!
![IMG_FE9425A837C0-1](https://user-images.githubusercontent.com/1127863/170169252-91401df2-c65f-4d0c-aae4-18ea7155ef80.jpeg)

**Comment by JmillsExpensify** [comment]
Ok great! I've just issued payments to all contributors, so I'm closing the issue. Don't hesitate to follow-up with any questions!
